### PR TITLE
use protostar for tests #5

### PR DESCRIPTION
**Pull request by ojetokun**

### CODE REVIEW

**Review by Pet3ris — COMMENTED** [review]
Cool – just needs a CI update now!

Would be great to find a fast way to install Protostar (possibly caching?) so that users still get very fast development loop w/ Protostar.

**Comment by ojetokun** [comment]
Considerations:
- Protostar tests depend on Scarb installation so a step was included for that

- Since Scarb comes with Cairo1, there is no need to have an additional step to install cairo1 independently so it was removed.
- Scarb release [0.2.0-alpha.1](https://github.com/software-mansion/scarb/releases/tag/v0.2.0-alpha.1) was used, which is the first version that supports cairo v1.0.0-rc0. 
- Also,
```
echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
source $GITHUB_PATH 
``` 
was used instead of 
```
echo "$HOME/.protostar/dist/protostar" >> $GITHUB_PATH
```
because the second just didn't work. The error being generated was `~/.protostar/dist/protostar is not a directory`. I suspect that it has something to do with the fact that the binary file is not in a `bin` folder.

@Pet3ris

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -13,15 +13,44 @@ jobs:
...
+          path: ~/.protostar
+          key: protostar-${{ env.PROTOSTAR_VERSION }}-${{ runner.os }}
+          restore-keys: protostar-${{ env.PROTOSTAR_VERSION }}-
 
+      - name: Protostar Setup
+        run: |
+          if [[ ! -d $HOME/.protostar/dist/protostar ]]; then
+            curl -L https://raw.githubusercontent.com/software-mansion/protostar/master/install.sh | bash -s -- -v $PROTOSTAR_VERSION
+          fi
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
```
Comment (Pet3ris) [inline .github/workflows/test.yaml:48]: why do we need this line?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -13,15 +13,44 @@ jobs:
...
+          source $GITHUB_PATH
```
Comment (Pet3ris) [inline .github/workflows/test.yaml:49]: why do we need this line? it wasn't needed before for the Cairo compiler?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -13,15 +13,44 @@ jobs:
...
+          fi
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
+          source $GITHUB_PATH          
+          protostar -v
+          
       - name: Test the code
-        run: cairo-test .
+        run: |
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
+          source $GITHUB_PATH
```
Comment (Pet3ris) [inline .github/workflows/test.yaml:55]: echo here

**Comment by Pet3ris** [comment]
> is not a directory

hmm, there's a rule with GitHub actions: "the currently running action cannot access the updated path variable"

any chance that's what is going on here? You need to update GITHUB_PATH / run in separate workflow steps?

**Review by Pet3ris — COMMENTED** [review]
Awesome, great to see CI working!

Just some questions on use of export/source rather than echo.

I think we'll just need to update README instructions on how to install Protostar/Scarb as opposed to installing cairo compiler directly.

**Comment by ojetokun** [comment]
> 



> > is not a directory
> 
> hmm, there's a rule with GitHub actions: "the currently running action cannot access the updated path variable"
> 
> any chance that's what is going on here? You need to update GITHUB_PATH / run in separate workflow steps?

Yup, thanks. Will make a push

File: .github/workflows/test.yaml (hunk 1 above)
Comment (ojetokun) [reply .github/workflows/test.yaml]: updated

File: .github/workflows/test.yaml (hunk 2 above)
Comment (ojetokun) [reply .github/workflows/test.yaml]: updated

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -13,15 +13,44 @@ jobs:
...
+            curl -L https://raw.githubusercontent.com/software-mansion/protostar/master/install.sh | bash -s -- -v $PROTOSTAR_VERSION
+          fi
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
+          source $GITHUB_PATH          
+          protostar -v
+          
       - name: Test the code
-        run: cairo-test .
+        run: |
+          echo "export PATH=\"\$PATH:$HOME/.protostar/dist/protostar\"" >> $GITHUB_PATH
```
[the remark this answers is not in the document]
Comment (ojetokun) [reply .github/workflows/test.yaml]: updated

**Review by Pet3ris — APPROVED** [review]
Wow - awesome work on the README, thanks for getting the details right!

**Pull request merged**